lib/client/ca_export_test.go

```diff
@@ -99,179 +110,402 @@ func TestExportAuthorities(t *testing.T) {
-
-				if exportSecrets {
-					exportFunc = ExportAuthoritiesSecrets
-					checkFunc = tt.assertSecrets
-				}
+		},
+		{
+			name: "fingerprint not found",
+			req: ExportAuthoritiesRequest{
+				AuthType:                   "user",
+				ExportAuthorityFingerprint: "fake fingerprint",
+			},
+			errorCheck: require.NoError,
+			assertNoSecrets: func(t *testing.T, output string) {
+				require.Empty(t, output)
+			},
+			assertSecrets: func(t *testing.T, output string) {
+				require.Empty(t, output)
+			},
+		},
+		{
+			name: "using compat version",
+			req: ExportAuthoritiesRequest{
+				AuthType:         "user",
+				UseCompatVersion: true,
+			},
+			errorCheck: require.NoError,
+			assertNoSecrets: func(t *testing.T, output string) {
+				// compat version (using 1.0) returns cert-authority to be used in the server
+				// even when asking for ssh authorized hosts / known hosts
+				require.Contains(t, output, "@cert-authority localcluster,*.localcluster ssh-rsa")
+			},
+			assertSecrets: validatePrivateKeyPEMFunc,
+		},
+		{
+			name: "db",
+			req: ExportAuthoritiesRequest{
+				AuthType: "db",
+			},
+			errorCheck:      require.NoError,
+			assertNoSecrets: validateTLSCertificatePEMFunc,
+			assertSecrets:   validatePrivateKeyPEMFunc,
+		},
+		{
+			name: "db-der",
+			req: ExportAuthoritiesRequest{
+				AuthType: "db-der",
+			},
+			errorCheck:      require.NoError,
+			assertNoSecrets: validateTLSCertificateDERFunc,
+			assertSecrets:   validatePrivateKeyDERFunc,
+		},
+		{
+			name: "db-client",
+			req: ExportAuthoritiesRequest{
+				AuthType: "db-client",
+			},
+			errorCheck:      require.NoError,
+			assertNoSecrets: validateTLSCertificatePEMFunc,
+			assertSecrets:   validatePrivateKeyPEMFunc,
+		},
+		{
+			name: "db-client-der",
+			req: ExportAuthoritiesRequest{
+				AuthType: "db-client-der",
+			},
+			errorCheck:      require.NoError,
+			assertNoSecrets: validateTLSCertificateDERFunc,
+			assertSecrets:   validatePrivateKeyDERFunc,
+		},
+	} {
+		runTest := func(
+			t *testing.T,
+			exportFunc func(context.Context, authclient.ClientI, ExportAuthoritiesRequest) ([]*ExportedAuthority, error),
+			assertFunc func(t *testing.T, output string),
+		) {
+			authorities, err := exportFunc(ctx, mockedAuthClient, tt.req)
+			tt.errorCheck(t, err)
+			if err != nil {
+				return
+			}
+
+			require.Len(t, authorities, 1, "exported authorities mismatch")
+			exported := string(authorities[0].Data)
+			assertFunc(t, exported)
+		}
+
+		runUnaryTest := func(
+			t *testing.T,
+			exportFunc func(context.Context, authclient.ClientI, ExportAuthoritiesRequest) (string, error),
+			assertFunc func(t *testing.T, output string),
+		) {
+			exported, err := exportFunc(ctx, mockedAuthClient, tt.req)
+			tt.errorCheck(t, err)
+			if err != nil {
+				return
+			}
+
+			assertFunc(t, exported)
+		}
+
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			t.Run(fmt.Sprintf("%s/ExportAllAuthorities", tt.name), func(t *testing.T) {
```

Comment:
I just noticed that the "%s/" is silly, this is already running in a subtest named after `tt.name`. I have some cleanups for master so I'll fix it there and backport to release branches.